core/templates/pages/collection-editor-page/services/collection-editor-state.service.ts

```diff
@@ -186,7 +188,9 @@ angular.module('oppia').factory('CollectionEditorStateService', [
             _collectionIsBeingSaved = false;
             if (successCallback) {
               successCallback();
+              $rootScope.$applyAsync();
             }
+            $rootScope.$applyAsync();
```

Comment:
Why we need this in both the place? i.e inside if and outside?

Also, should this applyAsync be called from directive (i.e, it should be a part of succesCallback not sure what pattern we follow) can you please check?

Reply:
You're right, I've moved it inside the successCallback and made sure that a successCallback is always passed (earlier I noticed that some calls to this function did not provide a callback so I added this.)